### PR TITLE
fix #214: AttributeDict does not support hasattr

### LINKED ISSUE #214: Can we redirect the output of terminal to .log or .txt file?
Hi,
 I want to redirect the output of terminal to a new file; they are too many and messy.
thanks!

### PR DESCRIPTION
**Pull request by jandom**
as per title

### CODE REVIEW

File: tests/test_utilities.py [hunk 1]
```diff
@@ -52,6 +52,11 @@ def test_dict_set(self):
         self.d["gargl"] = "blaster"
         assert self.d["gargl"] == "blaster"
 
+    def test_dict_hasttr(self):
```
Comment (orbeckst) [inline tests/test_utilities.py:55]: :-)
```suggestion
    def test_dict_hasattr(self):
```

**Review by orbeckst — CHANGES_REQUESTED** [review]
Thanks, just add an entry to CHANGES please.

(No idea why codecov doesn't see this one line covered that actually raises the AttributeError. But your tests seem to cover what's needed so I am ok with it.)

File: tests/test_utilities.py (hunk 1 above)
Comment (jandom) [reply tests/test_utilities.py]: good catch!

**Review by orbeckst — APPROVED** [review]

**Comment by orbeckst** [comment]
cheers!

**Comment by jandom** [comment]
Thanks for the speedy merge!

**Pull request merged**